### PR TITLE
Dbus Identity Property and Trust methods

### PR DESCRIPTION
**Pull request by bublath**
Giving this a first shot with just identities and trust - see discussion #1234

To be discussed:
Naming of methods and properties
Code for establishing trust is mostly copy&paste from TrustCommand.java - it might make sense to reuse it (but what is the best place?)

### CODE REVIEW

**Comment by bublath** [comment]
@AsamK Ping for feedback

**Comment by bublath** [comment]
@AsamK Are you still accepting PRs? It's a bit cumbersome to keep this patch in sync with your other changes. I can understand, if you don't have time for this currently, but would at least appreciate a quick feedback if you plan to review this at some time.
Thanks.

**Review by AsamK — COMMENTED** [review]
Thanks, I think this goes in the right direction.
I've now refactored the trust command in the master branch, that way you can have less copy&pasted code in the trust method.

File: src/main/java/org/asamk/Signal.java [hunk 1]
```diff
@@ -551,6 +551,48 @@ interface Group extends DBusInterface, Properties {
...
+    @DBusProperty(name = "id", type = String.class, access = DBusProperty.Access.READ)
+    @DBusProperty(name = "fingerprint", type = Byte[].class, access = DBusProperty.Access.READ)
+    @DBusProperty(name = "safetyNumber", type = String.class, access = DBusProperty.Access.READ)
+    @DBusProperty(name = "trustLevel", type = String.class, access = DBusProperty.Access.READ)
+    @DBusProperty(name = "addedDate", type = Integer.class, access = DBusProperty.Access.READ)
+    interface Identity extends DBusInterface, Properties {
+
+        void trust(String number) throws Error.Failure;
+
+        void trustWithKey(String number,String safetyNumber) throws Error.Failure;
```
[the remark this answers is not in the document]
Comment (bublath) [reply src/main/java/org/asamk/Signal.java:593]: ok

File: src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java [hunk 2]
```diff
@@ -1031,6 +1039,130 @@ private void exportObject(final DBusInterface object) {
...
+        }
+
+        @Override
+        public String getObjectPath() {
+            return getIdentityObjectPath(objectPath, identity.recipient().getLegacyIdentifier());
+        }
+
+        @Override
+        public void trust(String number) throws Error.Failure { 
+            trustWithKey(number,null);
```
[the remark this answers is not in the document]
Comment (bublath) [reply src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java:1095]: I'm struggling a bit here. I probably would need to implement the trustIdentityAllKeys() and trustIdentityVerified() in DbusManagerImpl.java like in ManagerImpl.java. However, to create the IdentityHelper it requires the context, which is not used for DbusManagerImpl. Any hint how to correctly get the IdentityHelper?

File: src/main/java/org/asamk/Signal.java [hunk 3]
```diff
@@ -551,6 +551,48 @@ interface Group extends DBusInterface, Properties {
...
+        public String getId() {
+            return id;
+        }
+
+        public String getName() {
+            return name;
+        }
+    }
+
+    @DBusProperty(name = "id", type = String.class, access = DBusProperty.Access.READ)
```
[the remark this answers is not in the document]
Comment (bublath) [reply src/main/java/org/asamk/Signal.java:584]: Just to confirm: I would use getLegacyIdentifier() for the number and getIdentifier() for uuid?

File: src/main/java/org/asamk/Signal.java [hunk 4]
```diff
@@ -551,6 +551,48 @@ interface Group extends DBusInterface, Properties {
...
+    @DBusProperty(name = "fingerprint", type = Byte[].class, access = DBusProperty.Access.READ)
+    @DBusProperty(name = "safetyNumber", type = String.class, access = DBusProperty.Access.READ)
```
[the remark this answers is not in the document]
Comment (bublath) [reply src/main/java/org/asamk/Signal.java:586]: ok

File: src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (AsamK) [reply src/main/java/org/asamk/signal/dbus/DbusSignalImpl.java:1095]: You don't have to implement it in the DbusManagerImpl, that's just required when using signal-cli as a DBus client.

Here you can just the code from the TrustCommand class, and only adapt the thrown exceptions:
```java
            try {
                final var res = m.trustIdentityAllKeys(recipient);
                if (!res) {
                    throw new UserErrorException("Failed to set the trust for this number, make sure the number is correct.");
                }
            } catch (UnregisteredRecipientException e) {
                throw new UserErrorException("The user " + e.getSender().getIdentifier() + " is not registered.");
            }
```
and
```java
            final IdentityVerificationCode verificationCode;
            try {
                verificationCode = IdentityVerificationCode.parse(safetyNumber);
            } catch (Exception e) {
                throw new UserErrorException(
                        "Safety number has invalid format, either specify the old hex fingerprint or the new safety number");
            }

            try {
                final var res = m.trustIdentityVerified(recipient, verificationCode);
                if (!res) {
                    throw new UserErrorException(
                            "Failed to set the trust for this number, make sure the number and the fingerprint/safety number are correct.");
                }
            } catch (UnregisteredRecipientException e) {
                throw new UserErrorException("The user " + e.getSender().getIdentifier() + " is not registered.");
            }
```

File: src/main/java/org/asamk/Signal.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (AsamK) [reply src/main/java/org/asamk/Signal.java:584]: You can use `number().orElse("")` and `uuid().map(UUID::toString).orElse("")` for those.

File: src/main/java/org/asamk/Signal.java [hunk 5]
```diff
@@ -551,6 +551,48 @@ interface Group extends DBusInterface, Properties {
         void enableLink(boolean requiresApproval) throws Error.Failure;
     }
```
[the remark this answers is not in the document]
Comment (bublath) [reply src/main/java/org/asamk/Signal.java:553]: ok

**Pull request closed without merging**